src/CreatorToken.sol

```diff
@@ -4,13 +4,15 @@ pragma solidity 0.8.20;
 import {ERC721} from "openzeppelin/token/ERC721/ERC721.sol";
 import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
 import {SafeERC20} from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import {Pausable} from "openzeppelin/security/Pausable.sol";
 
-contract CreatorToken is ERC721 {
+contract CreatorToken is ERC721, Pausable {
   using SafeERC20 for IERC20;
 
   error CreatorToken__MaxPaymentExceeded(uint256 _price, uint256 _maxPayment);
   error CreatorToken__CallerIsNotCreator(address _creator, address _caller);
   error CreatorToken__CallerIsNotAdmin(address _admin, address _caller);
+  error CreatorToken__CallerIsNotCreatorOrAdmin(address _creator, address _admin, address _caller);
```

Comment:
Let's collapse these three errors into one `CreatorToken_Unauthorized(bytes32 reason, address caller)`, where reason is either "not admin", "not creator", or "not creator or admin".

Reply:
updated in the same commit as above: https://github.com/showtime-xyz/creator-tokens/commit/310ca54770703a02b6bc34ae08dc1bab2b224243